frontend/src/app/features/auth/sign-up/sign-up.component.ts

```diff
@@ -0,0 +1,103 @@
+import { AbstractControl, FormControl, FormGroup, ValidatorFn, Validators } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { catchError, of } from 'rxjs';
+
+import { AuthService } from 'src/app/api/auth.service';
+import { Component } from '@angular/core';
+import { Store } from '@ngrx/store';
+import { register } from 'src/app/store/actions/login-page.actions';
+
+@Component({
+    selector: 'app-sign-up',
+    templateUrl: './sign-up.component.html',
+    styleUrls: ['./sign-up.component.sass']
+})
+export class SignUpComponent {
+
+
+    public signUpGroup: FormGroup;
+
+    constructor(
+        private authService: AuthService,
+        private store: Store,
+        private route: ActivatedRoute,
+        private router: Router
+    ) {
+        this.signUpGroup = this.buildSignUpGroup();
+    }
+
+    public signUpClick() {
+        this.authService.signUp({
+            firstName: this.signUpGroup.get('firstName')?.value!,
+            lastName: this.signUpGroup.get('lastName')?.value!,
+            email: this.signUpGroup.get('email')?.value!,
+            password: this.signUpGroup.get('password')?.value!
+        }).pipe(
+            catchError((resp) => {
+                if (resp.error === 'Account with this email already exists') {
+                    this.signUpGroup.get('email')?.setErrors({ emailTaken: true });
+                }
+                return of(null);
+            })
+        ).subscribe(
+            (user) => {
+                if (user == null) {
```

Comment:
I guess this null validatio is a result of catch error. Are there a way no to trigger subscribe when user is null?

Reply:
Used filter operator for ruling out null with map operator to fix type recognised in subscribe(without map it is still T | null)